### PR TITLE
rust: copy std dll into /usr/bin

### PR DESCRIPTION
**Pull request by ognevny**
fixes https://github.com/msys2/MSYS2-packages/issues/5784

### CODE REVIEW

File: rust/PKGBUILD [hunk 1]
```diff
@@ -98,6 +98,9 @@ package_rust() {
   rm components install.log manifest-* rust-installer-version uninstall.sh
   rm "${pkgdir}/etc/target-spec-json-schema.json"
 
+  # workaround for https://github.com/msys2/MSYS2-packages/issues/5784
+  cp "${pkgdir}"/usr/lib/rustlib/x86_64-pc-cygwin/lib/std-*.dll -t "${pkgdir}/usr/bin"
```
[the remark this answers is not in the document]
Comment (ognevny) [reply rust/PKGBUILD]: I believe that for windows-gnu it's a copied file, but didn't check rust sources

[the remark this answers is not in the document]
Comment (Berrysoft) [reply rust/PKGBUILD]: Oh, yes. It's a copied one.

[the remark this answers is not in the document]
Comment (500-internal-server-error) [reply rust/PKGBUILD]: Probably a good idea to hardlink it, so as not to waste space.

**Comment by oech3** [comment]
conflicting

**Comment by ognevny** [comment]
I know, I don't actually like this approach

**Comment by oech3** [comment]
any response from upstream?

**Comment by ognevny** [comment]
I don't know. I didn't open any issue in rust repo, because `-Cprefer-dynamic` is not recommended to use

**Comment by 500-internal-server-error** [comment]
[Upstream doesn't guarantee anything outside `$prefix/rustlib`](https://github.com/rust-lang/rust/issues/149050#issuecomment-3546395706), so we should merge this.

**Comment by oech3** [comment]
How many binaries linked against it?

**Comment by 500-internal-server-error** [comment]
Not many I imagine, even the one case I hit it was only in the test suite, not the build. Still would be nice not to hit it, though.

**Comment by Berrysoft** [comment]
Almost all rust binaries link std statically. It would be possible to link std dynamically if the std dll is in `/usr/bin`, but it would be massive for packagers, because every time the rust package updates, they need to update all binaries compiled by rust.

However, the proc macros use the std dll. `cargo` and `rustc` are responsible for their environments and the std dll doesn't have to be inside `/usr/bin`. Only some corner cases, like a test suite for some proc macro libs, might need the std dll in `PATH`.
